src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -734,52 +765,59 @@ private fun reexportedMacros(item: RsExternCrateItem): List<RsMacro>? {
     return exportingMacroNames.mapNotNull { nameToExportedMacro[it] }
 }
 
-private fun collectExportedMacros(
+private fun collectMacrosImportedWithUseItem(
     useItem: RsUseItem,
     exportingMacrosCrates: Map<String, RsMod>
 ): List<RsMacro> {
+    // We don't want to perform path resolution during macro resolve (because it can recursively perform
+    // macro resolve and so be slow and incorrect).
+    // We assume that macro can only be imported by 2-segment path (`foo::bar`), where the first segment
+    // is a name of the crate (may be aliased) and the second segment is the name of the macro.
     return buildList {
         val root = useItem.useSpeck ?: return@buildList
 
-        fun forEachRootUseSpeck(useSpeck: RsUseSpeck, consumer: (RsUseSpeck, RsPath) -> Unit) {
-            val path = useSpeck.path
-            if (path != null) {
-                consumer(useSpeck, path)
+        for ((crateName, macroName) in collect2segmentPaths(root)) {
+            val crateRoot = exportingMacrosCrates[crateName] as? RsFile
+                ?: findDependencyCrateByName(useItem, crateName)
```

Comment:
Do I understand correctly that `findDependencyCrateByName` is called when we don't have `extern crate` item?

Reply:
+